torchgeo/datasets/landsat.py

```diff
@@ -74,10 +74,10 @@ def __init__(
         Raises:
             FileNotFoundError: if no files are found in ``root``
         """
-        self.bands = bands if bands else self.all_bands
-        self.filename_glob = self.filename_glob.format(self.bands[0])
+        if bands:
+            self.filename_glob = self.filename_glob.format(bands[0])
```

Comment:
I don't think this will work, `self.filename_glob` always needs to be expanded or else we won't find any files during globbing. If `bands` is `[]` or `None`, could we just use `self.all_bands[0]`?